src/operators.rs

```diff
@@ -216,6 +220,13 @@ fn index_op(a: &FenderReference, b: &FenderReference) -> FenderReference {
     }
 }
 
+fn field_access(a: &FenderReference, b: &FenderReference) -> FenderReference {
+    match (&**a, &**b) {
+        (FenderValue::Struct(s), FenderValue::Int(i)) => s.data[i].dupe_ref(),
```

Comment:
We should make sure references behave as expected with struct fields

Reply:
from the checks Ive done so far they do